src/SubspacePreservingSparsification.jl

```diff
@@ -0,0 +1,17 @@
+module SubspacePreservingSparsification
+
+using LinearAlgebra
+using SparseArrays
+using SparseArrays: AbstractSparseMatrixCSC
+
+include("math_functions.jl")
+include("sparse_pattern.jl")
+include("bin_pattern.jl")
+include("compute.jl")
+
+export pinv_qr
+export p_norm_sparsity_matrix
+export bin_sparse_matrix!
```

Comment:
Do we need to export these functions for the user or are they just internal?

Reply:
actually I also would like to export the minimization function in a more general form too, you can calculate the sparsity pattern and the binning pattern independently from the minimization problem, the subspaces are preserved because auf the pattern, not because of the minimization problem, so maybe somebody would like to use an other optimization problem, I can write a bit more in the documentation.